src/app/dropdown-list/dropdown-list.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, forwardRef, Input } from '@angular/core'
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms'
+
+@Component({
+  selector: 'app-dropdown-list',
+  templateUrl: './dropdown-list.component.html',
+  styleUrls: ['./dropdown-list.component.sass'],
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => DropdownListComponent),
+      multi: true
+    }
+  ]
+})
+export class DropdownListComponent implements ControlValueAccessor {
+  constructor () { }
+  @Input()
+  options: string[] = ['option 1', 'option 2', 'option 3']
+
+  @Input()
+  multiSelect: boolean = false;
+
+  onChange: any = () => {}
+  onTouch: any = () => {}
+  value: (string[] | string) = ['']
+  toggleMulti (val: string) {
+    if (!Array.isArray(this.value)) return
```

Comment:
What do you think about this one? I want to return string or string[] depending on multiSelect = true/false.